vite.config.development.ts

```diff
@@ -1,11 +1,36 @@
+import { readFileSync as read } from 'node:fs'
 import { defineConfig, mergeConfig, type UserConfigFn, type UserConfig } from 'vite'
 
 import { config as prodConfig } from './vite.config.production'
+const version = JSON.parse(read('./package.json').toString()).version
 // https://vitejs.dev/config/
 export const config: UserConfigFn = (env) => {
   return mergeConfig(
     prodConfig(env),
```

Comment:
unrelated, but why is there a `prodConfig` in a file called `development.ts`?

Reply:
I think that `vite.config.production` is the base and in `vite.config.development` some pieces are overridden: [vite/mergeConfig](https://vite.dev/guide/api-javascript.html#mergeconfig) 🤔 not sure if it's worth to create another config called `base` 🙂

Reply:
I'd say if you put something prod specific in `production` and don't overwrite it in `dev` then we're in trouble because we might be sending some data cross env. I'd be heavily in favour of creating `base`.

Reply:
This is a bit of a weird one, and there are things to keep in mind mostly point 1 here:

1. We can't have a file called `vite.config.ts` in the root. If we do, vitepress will use it for its config automatically (ouch) and we don't want that happening. This is the biggest reason it got renamed to `.production`, at the time we just needed a different name, any different name, seeing as we already had a `.development` it got moved to `vite.config.production`. `.base` would also be fine for a name.
2. FYI `vite.development` was added incase we needed it (and I'm pretty sure we don't now) and `vite.config.preview` doesn't use the `production` file anyway from what I remember.
3. We have a longer running task to move all our config into a shareable JS package at `@kumahq/config`, and I was hoping to reorganize the vite config into that also, at which point all this would change. Next steps for this long running task is moving our repository to use npm workspaces though, so we are just waiting for a good time to do that before moving all these config files around.

I think the name will end up being `vite.config.app.ts`, which will source its config from `@kumahq/config`, then we'll have a `vite.config.e2e.ts` or something that will also source its config from `@kumahq/config` (and vitepress's config will also come from `@kumahq/config`.

In the meantime if we want to change it to something else I'm fine with that and would be happy to 👍 a PR, but take into account it will probably change again once we have `@kumahq/config` available everywhere.

Reply:
I think we should leave this as is _for now_ i.e. in this PR.

I agree that the `.production` suffix isn't a great name and can be confusing, but seeing as all our config is going to be moved around pretty soon (hopefully) and the current state has been around for quite some time, I'd say lets leave as is in this PR and when we move the config files around lets make sure we don't keep this naming scheme.

I'm gonna resolve this convo, but let me know if you want to reopen